runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxPanelManager.java

```diff
@@ -0,0 +1,191 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.game.chatbox;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import java.util.Arrays;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.ScriptID;
+import net.runelite.api.VarClientInt;
+import net.runelite.api.events.ScriptCallbackEvent;
+import net.runelite.api.widgets.JavaScriptCallback;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.input.KeyListener;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.input.MouseListener;
+import net.runelite.client.input.MouseManager;
+import net.runelite.client.input.MouseWheelListener;
+
+@Singleton
+@Slf4j
+public class ChatboxPanelManager
+{
+	private final Client client;
+	private final ClientThread clientThread;
+	private final EventBus eventBus;
+
+	private final KeyManager keyManager;
+	private final MouseManager mouseManager;
+
+	private final Provider<ChatboxTextMenuInput> chatboxTextMenuInputProvider;
+	private final Provider<ChatboxTextInput> chatboxTextInputProvider;
+
+	@Getter
+	private ChatboxInput currentInput = null;
+
+	@Inject
+	private ChatboxPanelManager(EventBus eventBus, Client client, ClientThread clientThread,
+		KeyManager keyManager, MouseManager mouseManager,
+		Provider<ChatboxTextMenuInput> chatboxTextMenuInputProvider, Provider<ChatboxTextInput> chatboxTextInputProvider)
+	{
+		this.client = client;
+		this.clientThread = clientThread;
+		this.eventBus = eventBus;
+
+		this.keyManager = keyManager;
+		this.mouseManager = mouseManager;
+
+		this.chatboxTextMenuInputProvider = chatboxTextMenuInputProvider;
+		this.chatboxTextInputProvider = chatboxTextInputProvider;
+
+		eventBus.register(this);
+	}
+
+	public void close()
+	{
+		clientThread.invokeLater(this::unsafeCloseInput);
+	}
+
+	void unsafeCloseInput()
+	{
+		client.runScript(
+			ScriptID.RESET_CHATBOX_INPUT,
+			0,
+			1
+		);
+	}
+
+	void unsafeOpenInput(ChatboxInput input)
+	{
+		client.runScript(ScriptID.CLEAR_CHATBOX_PANEL);
+
+		eventBus.register(input);
+		if (input instanceof KeyListener)
+		{
+			keyManager.registerKeyListener((KeyListener) input);
+		}
+		if (input instanceof MouseListener)
+		{
+			mouseManager.registerMouseListener((MouseListener) input);
+		}
+		if (input instanceof MouseWheelListener)
+		{
+			mouseManager.registerMouseWheelListener((MouseWheelListener) input);
+		}
+
+		currentInput = input;
+		client.setVar(VarClientInt.INPUT_TYPE, -3);
```

Comment:
What is -3? I believe we have now constant class for input type, probably put this -3 there